### PR TITLE
Include project items in masthead search

### LINKED ISSUE #2591: Quick search should include project image streams and templates
The browse catalog from project shows project templates and image streams. Quick search from the project masthead does not, however. This means that you get inconsistent results.

openshift web console 2017-12-12 15-04-47

/kind bug
/priority p2
/assign

@jeff-phillips-18 @jwforres FYI

### PR DESCRIPTION
**Pull request by spadgett**
Show templates and image streams for the current project in the catalog
search from the project header.

Fixes #2591

### CODE REVIEW

**Comment by spadgett** [comment]
@dtaylor113 PTAL

File: app/scripts/directives/nav.js [hunk 1]
```diff
@@ -342,8 +343,14 @@ angular.module('openshiftConsole')
...
-                  $scope.catalogItems = items;
+                var catalogItemsPromise = CatalogService.getCatalogItems().then(function(items) {
+                  catalogItems = items;
+                });
+                var projectItemsPromise = Catalog.getProjectCatalogItems(currentProjectName).then(_.spread(function(catalogServiceItems) {
+                  projectItems = catalogServiceItems;
+                }));
+                $q.all([catalogItemsPromise, projectItemsPromise]).then(function() {
+                  $scope.catalogItems = Catalog.sortCatalogItems(_.concat(catalogItems, projectItems));
                 });
```
Comment (dtaylor113) [inline app/scripts/directives/nav.js]: What happens if there is a failure getting catalogItems and/or projectItems?   There doesn't seem to be a failure/error block.

Comment (spadgett) [reply app/scripts/directives/nav.js]: An error is displayed by data service. Right now catalog service in origin-web-catalog doesn't tell data service to suppress error notifications.

**Comment by spadgett** [comment]
@dtaylor113 Any other concerns with the PR?

If I add errors where you mention, errors would be toasted twice.

**Pull request merged**